api/v1/storagecluster_types.go

```diff
@@ -137,11 +140,41 @@ type ManageCephObjectStoreUsers struct {
 	ReconcileStrategy string `json:"reconcileStrategy,omitempty"`
 }
 
+// ExternalStorageKind specifies a kind of the external storage
+type ExternalStorageKind string
+
+const (
+	// KindOCS specifies a "ocs" kind of the external storage
+	KindOCS ExternalStorageKind = "ocs"
+
+	// KindRHCS specifies a "rhcs" kind of the external storage
+	KindRHCS ExternalStorageKind = "rhcs"
+)
+
 // ExternalStorageClusterSpec defines the spec of the external Storage Cluster
 // to be connected to the local cluster
 type ExternalStorageClusterSpec struct {
-	// +optional
+	//+kubebuilder:default:=true
+	// Enable enables the external mode
 	Enable bool `json:"enable,omitempty"`
+
+	//+kubebuilder:default:=rhcs
+	// StorageProviderKind describes the kind of the external storage
+	StorageProviderKind ExternalStorageKind `json:"storageProviderKind,omitempty"`
+
+	// ConnectionString describes the blob of provider
+	ConnectionString string `json:"connectionString,omitempty"`
+
+	//+kubebuilder:default:="1Ti"
+	// RequestedCapacity defines the requested storage on the provider
+	RequestedCapacity resource.Quantity `json:"requestedCapacity,omitempty"`
+}
+
+// ExternalStorageClusterStatus defines the status of the external Storage Cluster
+// to be connected to the local cluster
+type ExternalStorageClusterStatus struct {
+	// ActualCapacity defines the actual storage provided by the provider
```

Comment:
Is this the capacity of the Provider OCS or Consumer OCS?

Reply:
It will be the capacity that is provided to the consumer by the provider. which means consumer capacity.

Reply:
@iamniting It think the name is misleading, Can we please rename form `ActualCapacity` to `GrantedCapacity`?